### PR TITLE
Modelchat worker qual

### PR DESCRIPTION
**Pull request by mojtaba-komeili**
**Patch description**
* Added allow list qualification to the model chat

### CODE REVIEW

**Comment by klshuster** [comment]
probably good to have @EricMichaelSmith's review here but it all looks good to me

File: parlai/crowdsourcing/tasks/model_chat/README.md [hunk 1]
```diff
@@ -54,6 +54,8 @@ Some options for running human+model image chat are as follows:
...
 - `mephisto.blueprint.evals_per_image_model_combo`: the maximum number of conversations collected for each combination of image and model. For instance, if this is set to 3 and your 2 models are `model_1` and `model_2`, each image will have 6 conversations collected about it, 3 with `model_1` and 3 with `model_2`.
 - `mephisto.blueprint.world_file`: the path to the Python module containing the class definition for the chat World, used for setting the logic for each turn of the conversation, when to end the conversation, actions upon shutdown, etc. (The onboarding World, if it exists, will be defined in this module as well.) Modify this value if you would like to write your own World class without having to create a new Blueprint class.
+- `allowed_worker_qualification`: use this qualification to restrict the task to an exclusive list of workers. Not that you must first grant this qualification to the workers who are allowed to participate.
```
Comment (EricMichaelSmith) [inline parlai/crowdsourcing/tasks/model_chat/README.md:57]: nit: typo in "Note"

File: parlai/crowdsourcing/tasks/model_chat/impl.py [hunk 2]
```diff
@@ -7,14 +7,17 @@
...
 from mephisto.tools.scripts import load_db_and_process_config
-from omegaconf import DictConfig, OmegaConf
+from mephisto.data_model.qualification import QUAL_EXISTS
+from mephisto.utils.qualifications import make_qualification_dict
 
 from parlai.crowdsourcing.utils.mturk import soft_block_mturk_workers
 from parlai.crowdsourcing.tasks.model_chat.model_chat_blueprint import (
     SharedModelChatTaskState,
 )
+import parlai.utils.logging as logging
```
Comment (EricMichaelSmith) [inline parlai/crowdsourcing/tasks/model_chat/impl.py:20]: nit: lint says this is unused

File: parlai/crowdsourcing/tasks/model_chat/impl.py [hunk 3]
```diff
@@ -47,7 +50,17 @@ def run_task(cfg: DictConfig, task_directory: str, world_module=None):
     soft_block_mturk_workers(cfg=cfg, db=db, soft_block_qual_name=soft_block_qual_name)
 
     # Init
-    shared_state = SharedModelChatTaskState(world_module=world_module)
+    if cfg.mephisto.blueprint.allowed_worker_qualification is not None:
```
Comment (EricMichaelSmith) [inline parlai/crowdsourcing/tasks/model_chat/impl.py:53]: Should we add a TODO to bubble this functionality up to all ParlAI Mephisto tasks in the future?

**Review by EricMichaelSmith — APPROVED** [review]
Wow, we've had blocklisting forever but not allowlisting, great to have! Minor comments

**Pull request merged**